docs/workflow/ci/ci_health.md

```diff
@@ -0,0 +1,167 @@
+
+Opening the [runtime-outerloop Pipeline Pass Rate](https://dev.azure.com/dnceng/public/_pipeline/analytics/stageawareoutcome?definitionId=98&contextType=build) there is a presentation of a line graph of the end to end success rate for the pipeline over time.
+
+The next graph attempts to show what is failing in bar graph and give a small insight into what is generally failing.
+
+The last graph is the most interesting for finding specific issues. As of writing the `Top 10 failing tasks` are:
+
+*Note* that any one of these buckets can include random one off infrastructure failures or systimatic Azure Dev Ops failures. For example the build bucket can include issues like:
+
+>  fips.c(143): OpenSSL internal error
+
+**Failure Buckets**
+
+1. Default\Send tests to Helix
+    - This set can be one of two problems. Either we have tests that failed or helix has failed with some infrastructure issue.
+2. Default\Build product
+    - Build related failures
+3. Default\Build managed test components
+    - Build failures while build the managed components of our tests
+4. Default\Initialize containers
+    - This is an azure dev ops infrastructure issue. It manifests while setting up the docker container and fails to start the environemnt correctly.
+5. Send Helix End Telemetry
+    - This is a helix infrastructure issue
+6. Default\Build native test components
+    - This is a failure in the native test build. Generally this is an Azure Dev Ops issue because we do little work inside this step.
+7. Default\Download product build
+    - Generally this is an Azure Dev Ops issue because we do little work inside this step.
+8. Default\Unsize GIT Repository
+    - Generally this is an Azure Dev Ops issue because we do little work inside this step.
+9. Default\Component Detection
+    - This is an azure dev ops issue
+
+Below each of these buckets are tabs which show individual runs which can be drilled through to find specific instances of each failure.
+
+**Test Pass rate**
+
+This drill through is extremely useful for finding individual flakey tests. Our idealogy is that tests should be 100% reliable. Tests which appear on this list should be disabled **and** fixed. As even a small amount of unreliability in the tests will equate to a significant percentage of pipeline failiure percentage.
```

Comment:
According to wiktionary "idealogy" is a misspelling of "ideology" even though I wonder whether it should rather be something like "principle", "guidance", "rule" (or maybe "rule of thumb").

Reply:
Typo - "failiure" -> "failure"

Reply:
And the wording "significant percentage of pipeline failure percentage" sounds somewhat weirdly repetitive, I think we can drop the second "percentage".